src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_PatternSwitchStatement.cs

```diff
@@ -60,13 +60,20 @@ private BoundStatement MakeLoweredForm(BoundPatternSwitchStatement node)
                 var expression = _localRewriter.VisitExpression(node.Expression);
                 var result = ArrayBuilder<BoundStatement>.GetInstance();
 
-                if (expression.ConstantValue == null)
+                // EnC: We need to insert a hidden sequence point to handle function remapping in case
+                // the containing method is edited while methods invoked in the expression are being executed.
+                if (!node.WasCompilerGenerated && _localRewriter.Instrument)
                 {
-                    // EnC: We need to insert a hidden sequence point to handle function remapping in case
-                    // the containing method is edited while methods invoked in the expression are being executed.
-                    if (!node.WasCompilerGenerated && _localRewriter.Instrument)
+                    var instrumentedExpression = _localRewriter._instrumenter.InstrumentSwitchStatementExpression(node, expression, _factory);
+                    if (expression.ConstantValue == null)
                     {
-                        expression = _localRewriter._instrumenter.InstrumentSwitchStatementExpression(node, expression, _factory);
+                        expression = instrumentedExpression;
+                    }
+                    else
+                    {
+                        // If the expression is a constant, we leave it alone (the decision tree lowering code needs
+                        // to see that constant). But we add an additional leading statement with the instrumented expression.
+                        result.Add(_factory.ExpressionStatement(instrumentedExpression));
```

Comment:
Only the debug instrumentation is affected (it inserts a hidden sequence point). The other instrumentation forms ignore the switch expression. In other words, there is no instrumentation test that would observe a change in behavior.